nebula_decoders/src/nebula_decoders_continental/decoders/continental_ars548_decoder.cpp

```diff
@@ -0,0 +1,574 @@
+                                                         : "INVALID VALUE";
+
+  radar_status_.characteristic_speed_status =
+    sensor_status_packet.characteristic_speed_status == 0   ? "VDY_OK"
+    : sensor_status_packet.characteristic_speed_status == 1 ? "VDY_NOTOK"
+                                                            : "INVALID VALUE";
+
+  if (sensor_status_packet.radar_status == 0) {
+    radar_status_.radar_status = "STATE_INIT";
+  } else if (sensor_status_packet.radar_status == 1) {
+    radar_status_.radar_status = "STATE_OK";
+  } else if (sensor_status_packet.radar_status == 2) {
+    radar_status_.radar_status = "STATE_INVALID";
+  } else {
+    radar_status_.radar_status = "INVALID VALUE";
+  }
+
+  if (sensor_status_packet.voltage_status == 0) {
+    radar_status_.voltage_status = "Ok";
+  }
+  if (sensor_status_packet.voltage_status & 0x01) {
+    radar_status_.voltage_status += "Current under voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x02) {
+    radar_status_.voltage_status = "Past under voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x03) {
+    radar_status_.voltage_status = "Current over voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x04) {
+    radar_status_.voltage_status = "Past over voltage";
+  }
+
+  if (sensor_status_packet.temperature_status == 0) {
+    radar_status_.temperature_status = "Ok";
+  }
+  if (sensor_status_packet.temperature_status & 0x01) {
+    radar_status_.temperature_status += "Current under temperature";
+  }
+  if (sensor_status_packet.temperature_status & 0x02) {
+    radar_status_.temperature_status += "Past under temperature";
+  }
+  if (sensor_status_packet.temperature_status & 0x03) {
+    radar_status_.temperature_status += "Current over temperature";
+  }
+  if (sensor_status_packet.temperature_status & 0x04) {
+    radar_status_.temperature_status += "Past over temperature";
+  }
+
+  const uint8_t & blockage_status0 = sensor_status_packet.blockage_status & 0x0f;
+  const uint8_t & blockage_status1 = (sensor_status_packet.blockage_status & 0xf0) >> 4;
+
+  if (blockage_status0 == 0) {
+    radar_status_.blockage_status = "Blind";
+  } else if (blockage_status0 == 1) {
+    radar_status_.blockage_status = "High";
+  } else if (blockage_status0 == 2) {
+    radar_status_.blockage_status = "Mid";
+  } else if (blockage_status0 == 3) {
+    radar_status_.blockage_status = "Low";
+  } else if (blockage_status0 == 4) {
+    radar_status_.blockage_status = "None";
+  } else {
+    radar_status_.blockage_status = "INVALID VALUE";
+  }
+
+  if (blockage_status1 == 0) {
+    radar_status_.blockage_status += ". Self test failed";
+  } else if (blockage_status1 == 1) {
+    radar_status_.blockage_status += ". Self test passed";
+  } else if (blockage_status1 == 2) {
+    radar_status_.blockage_status += ". Self test ongoing";
+  } else {
+    radar_status_.blockage_status += ". Invalid self test";
+  }
```

Comment:
These would be clearer as switch statements

Reply:
Addressed in [800b351](https://github.com/tier4/nebula/pull/123/commits/800b351ad5c941b14e31507c4cd59eb6b42ddd4a)